cdi-core/src/main/java/com/linkedin/cdi/connection/S3Connection.java

```diff
@@ -59,6 +54,7 @@
  */
 public class S3Connection extends MultistageConnection {
   private static final Logger LOG = LoggerFactory.getLogger(S3Connection.class);
+  private static final String UPLOAD_S3_KEY = "s3key";
```

Comment:
Please use a more descriptive and unique name here. s3key is very general and could conflict with other use cases which happen to use the same name

Reply:
This is only for this context though, how about `ms.target.s3.key`? similar to `ms.extractor.target.file.name`

Reply:
I'll change to upload.s3.key, this won't conflict because it's in the context of the upload secondary input

Reply:
yeah uploadS3Key is fine. If you use s3key, it is used by other flows as well so other flow may end up enabling this feature unexpectedly